src/Microsoft.AspNetCore.Authentication.JwtBearer/JwtBearerHandler.cs

```diff
@@ -103,7 +106,8 @@ internal class JwtBearerHandler : AuthenticationHandler<JwtBearerOptions>
                             Logger.TokenValidationFailed(token, ex);
 
                             // Refresh the configuration for exceptions that may be caused by key rollovers. The user can also request a refresh in the event.
-                            if (Options.RefreshOnIssuerKeyNotFound && ex.GetType().Equals(typeof(SecurityTokenSignatureKeyNotFoundException)))
+                            if (Options.RefreshOnIssuerKeyNotFound && Options.ConfigurationManager != null
+                                && ex.GetType().Equals(typeof(SecurityTokenSignatureKeyNotFoundException)))
```

Comment:
this will skip derived types of SecurityTokenSignatureKeyNotFoundException. if it matters to you, you could do 'is' or 'as'.